iothub/device/src/ClientFactory.cs

```diff
@@ -131,19 +130,6 @@ private static void EnsureOptionsIsSetup(X509Certificate2 cert, ref IotHubClient
             {
                 options.FileUploadTransportSettings = new();
             }
-
-            if (cert != null)
```

Comment:
We don't need to propagate the certificate information this way anymore. Certs will be stored in IotHubConnectionCredentials and will be propagated to the transport layer.